pdr_backend/lake/test/test_gql_data_factory.py

```diff
@@ -298,3 +331,129 @@ def test_do_subgraph_fetch_stop_loop_when_restarting_fetch(
     assert len(pds.query_data("SELECT * FROM {}".format(temp_table_name))) == 3
 
     assert "Fetched" in caplog.text
+
+
+def test_get_etl_st_fin(_gql_datafactory_etl_predictions_df, tmpdir):
+    st_timestr = "2023-01-03"
+    fin_timestr = "2024-05-05"
+    ppss = mock_ppss(
+        ["binance BTC/USDT c 5m"],
+        "sapphire-mainnet",
+        str(tmpdir),
+        st_timestr=st_timestr,
+        fin_timestr=fin_timestr,
+    )
+
+    gql_data_factory = GQLDataFactory(ppss)
+    gql_data_factory.record_config["gql_tables"] = ["pdr_predictions"]
+
+    st_ut, fin_ut = gql_data_factory.get_etl_st_fin()
+
+    assert st_ut is None
+    assert fin_ut is None
+
+    pds = PersistentDataStore(ppss.lake_ss.lake_dir)
+    # Add some predictions
+    pds.drop_table(get_table_name("pdr_predictions", TableType.TEMP))
+
+    pds.insert_to_table(
+        _gql_datafactory_etl_predictions_df,
+        get_table_name("pdr_predictions", TableType.TEMP),
+    )
+
+    gql_data_factory._move_from_temp_tables_to_live()
+
+    st_ut, fin_ut = gql_data_factory.get_etl_st_fin()
+
+    assert st_ut.to_seconds() == 1672704000
+    assert fin_ut.to_seconds() == 1699300700
+
+def test_prepare_temp_table(
+        _gql_datafactory_etl_predictions_df,
+        _gql_datafactory_etl_predictions_df_second_part,
+        tmpdir
+    ):
+    st_timestr = "2023-06-01"
+    fin_timestr = "2024-05-05"
+    ppss = mock_ppss(
+        ["binance BTC/USDT c 5m"],
+        "sapphire-mainnet",
+        str(tmpdir),
+        st_timestr=st_timestr,
+        fin_timestr=fin_timestr,
+    )
+
+    gql_data_factory = GQLDataFactory(ppss)
+    gql_data_factory.record_config["gql_tables"] = ["pdr_predictions"]
+
+    pds = PersistentDataStore(ppss.lake_ss.lake_dir)
+    csvds = CSVDataStore(ppss.lake_ss.lake_dir)
+    # Add some predictions
+    pds.drop_table(get_table_name("pdr_predictions", TableType.TEMP))
+    csvds.delete("pdr_predictions")
```

Comment:
i don't get this...
1. pytest creates a different tmpdir for every test
2. you don't need to delete anything, there is nothing there
3. there is now a function to delete csvs which is only used by this test